### ISSUE TITLE
Unsupported BitsPerSample for SampleFormat 3/Floating Point (expected 16/32): 64

### ISSUE THREAD

**Issue by sroettering** [haraldk/TwelveMonkeys#948]
**Describe the bug**
When reading the attached tiff file I get the error already stated in the issue title. Here is the complete stacktrace:

```
javax.imageio.IIOException: Unsupported BitsPerSample for SampleFormat 3/Floating Point (expected 16/32): 64
	at com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReader.getDataType(TIFFImageReader.java:819)
	at com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReader.getRawImageType(TIFFImageReader.java:459)
	at com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReader.getImageTypes(TIFFImageReader.java:909)
	at com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReader.read(TIFFImageReader.java:938)
	at java.desktop/javax.imageio.ImageIO.read(ImageIO.java:1466)
	at java.desktop/javax.imageio.ImageIO.read(ImageIO.java:1363)
	at com.x.x.x.x.png.PngRendererAccessor.renderGrayScaleGeoTiff(PngRendererAccessor.java:71)
```

Image is in this zip file since Github says it does not support tiff.
[analysisTiff.tif.zip](https://github.com/haraldk/TwelveMonkeys/files/15349512/analysisTiff.tif.zip)

**Version information**
1. The version of the TwelveMonkeys ImageIO library in use.
```
dependencies {
  implementation 'com.twelvemonkeys.imageio:imageio-jpeg:3.10.1'
  implementation 'com.twelvemonkeys.imageio:imageio-tiff:3.10.1'
  implementation 'com.twelvemonkeys.contrib:contrib:3.10.0'
}
```

**To Reproduce**

The following code already throws the error:
```
ImageIO.read(tiffInputStream)
```

**Expected behavior**
The attached file can be read without problems.

I just added the twelvemonkeys tiff plugin today and one of our tests broke because the tiff could not be read anymore. Without the plugin ImageIO was able to read it. However, I need the plugin for another usecase to work correctly.

Bit of context that may be useful: In the agricultural sector (geo)tiffs are often used to transport/encode information that might not always be used to be viewed directly but analyzed or transformed further. As is in this case I need to create a grayscale image from the input tiff.

**Comment by haraldk** [comment]
Hi @sroettering  

Sorry about the delayed response, I've been busy celebrating national holidays and enjoying spring time away from the IDE... 😉 

Thanks for reporting! I'm not sure if Java 2D has a good way of storing 64 bit FP (Java: `double`) samples... But if the JDK TIFF plugin supports that, there must be a way. 😀 

I'll look into it!

**Comment by haraldk** [comment]
I think I figured it out, just needed a few extra methods, but the main code seems to work fine for 64 bit FP. 

However, the resulting image doesn't look exactly like the one from the JDK TIFF plugin... Unfortunately, I have no other tools that can process/display this image and no knowledge of what it's supposed to resemble. But at least it can be read:

JDK TIFF plugin:
![image](https://github.com/haraldk/TwelveMonkeys/assets/110299/ffb6ec8b-5064-4f70-8c43-f3e7a25e8938)

TwelveMonkeys TIFF plugin:
![image](https://github.com/haraldk/TwelveMonkeys/assets/110299/60c69cf8-69af-4dd5-932b-5728256f1b00)

**Comment by sroettering** [comment]
Wow great to hear! I will give it a try when I find the time, thanks.

**Comment by haraldk** [comment]
Fixed in #948
